required/latex-lab/latex-lab-math.dtx

```diff
@@ -1349,25 +1349,25 @@
 % \subsection{Content grabbing}
 %
 % \begin{macro}{\@@_grab_dollar:w}
-%   Grab up to a single |$|, for inline math mode, suppressing
-%   any processing if the token is \tn{m@th} found in the content.
-%
-%    \fmi{what's that test doing?}
-%
-%    \car{It is some kind of fix, to avoid the remote
-%      possibility that the math is empty, making the code
-%      produce an unwanted \texttt{\$\$}.}
-%
-%\car{cf.~the code for this in \cs{@ensuredmath}}
-%
-%\car{It is harmless but unnecessary in the dollardollar grabbing below.}
-%
+% \begin{macro}{\@@_grab_dollar:n}
+%   Top-level function to handle grabbing of inline math mode delimited by
+%   |$| tokens. We provide two different ways to do that: a token-by-token
+%   one that can be used everywhere, and a fast delimited one that does not
+%   work anywhere that the end |$| token may be hidden, most obviously in
+%   tabulars. The function here is therefore set up as a variable starting
+%   point.
 %    \begin{macrocode}
-\cs_new_protected:Npn \@@_grab_dollar:w % $
-  #1 $
+\cs_new_protected:Npn \@@_grab_dollar:w { \@@_grab_dollar_delim:w }
+%    \end{macrocode}
+%   After grabbing inline math material, there is again common processing
+%   independent of mechanism of collection.
+%    \begin{macrocode}
+\cs_new_protected:Npn \@@_grab_dollar:n #1
   {
 %    \end{macrocode}
 %    \fmi{what's that test doing?}
```

Comment:
so my question should be dropped I guess. But is that actually a common case that warrants testing for? (not that it hurt)

Reply:
I've dropped the comment - I think the cost of the test is not high and should mean we don't need to worry about empty structures from 'fake' math or similar, so would favour keeping it.